src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas2/StatusUpdateServiceTest.kt

```diff
@@ -429,4 +445,309 @@ class StatusUpdateServiceTest {
+            .withAssessment(assessment)
+            .withAssessor(assessor)
+            .withStatusId(activeStatusWithDetail.id)
+            .produce()
+
+          val cas2StatusUpdateDetailEntity = Cas2StatusUpdateDetailEntity(
+            id = UUID.randomUUID(),
+            statusUpdate = cas2StatusUpdateEntity,
+            statusDetailId = statusDetail.id,
+            label = statusDetail.label,
+          )
+
+          every { mockAssessmentRepository.findByIdOrNull(assessment.id) } answers
+            {
+              assessment
+            }
+
+          every { mockStatusUpdateRepository.save(any()) } answers
+            {
+              cas2StatusUpdateEntity
+            }
+
+          every { mockStatusUpdateDetailRepository.save(any()) } answers
+            {
+              cas2StatusUpdateDetailEntity
+            }
+
+          every { mockDomainEventService.saveCas2ApplicationStatusUpdatedDomainEvent(any()) } answers { }
+
+          every { mockNotifyConfig.templates.cas2ApplicationStatusUpdated } returns "abc123"
+
+          every {
+            mockEmailNotificationService.sendCas2Email(
+              recipientEmailAddress = applicant.email!!,
+              templateId = "abc123",
+              personalisation = mapOf(
+                "applicationStatus" to cas2StatusUpdateEntity.label,
+                "dateStatusChanged" to cas2StatusUpdateEntity.createdAt.toLocalDate().toCas2UiFormat(),
+                "timeStatusChanged" to cas2StatusUpdateEntity.createdAt.toCas2UiFormattedHourOfDay(),
+                "applicationType" to "Home Detention Curfew (HDC)",
+                "nomsNumber" to "NOMSABC",
+                "applicationUrl" to "http://example.com/application/$applicationId/overview",
+              ),
+            )
+          } just Runs
+        }
+
+        @Test
+        fun `saves a status update entity with detail and emits a domain event`() {
+          every { mockStatusTransformer.transformStatusDetailListToDetailItemList(listOf(statusDetail)) } returns listOf(
+            Cas2StatusDetail("exampleStatusDetail", ""),
+          )
+
+          statusUpdateService.createForAssessment(
+            assessmentId = assessment.id,
+            statusUpdate = applicationStatusUpdateWithDetail,
+            assessor = assessor,
+          )
+
+          verify {
+            mockStatusUpdateRepository.save(
+              match {
+                it.statusId == activeStatusWithDetail.id
+              },
+            )
+          }
+
+          verify {
+            mockStatusUpdateDetailRepository.save(
+              match {
+                it.statusDetailId == statusDetail.id &&
+                  it.label == statusDetail.label
+              },
+            )
+          }
+
+          verify {
+            mockDomainEventService.saveCas2ApplicationStatusUpdatedDomainEvent(
+              match {
+                it.crn == "CRN123" &&
+                  it.applicationId == applicationId &&
```

Comment:
AssessmentId check?

Reply:
as above - we don't pass in assessment  to the domain event. It's a good question though whether we *should* pass it in to the domain event. These are consumed by Delius, so changing their format would be a whole story in itself, which I will add to the docs. 

 I think we have enough coverage to check the assessment id has been saved as I'm checking the below on a successful save for both `create` and `createForAssessment`.
 ```
         verify {
          mockStatusUpdateRepository.save(
            match {
              it.assessment!!.id == assessment.id
            },
          )
        }
```

Reply:
Ok so no requirement for it to be added at the moment.  Always good to have this requirement pulled by someone else rather than pushing it to them - good Kanban!